enhancements/sig-architecture/105-aws-iam-registration/README.md

```diff
@@ -0,0 +1,464 @@
+```
+hub_account_id: <hub-account-id>
+hub_cluster_name: <hub-cluster-name>
+managed_cluster_name: <managed-cluster-name>
+```
+
+**Note:** The _md5-hash-unique-identifier_ should be calculated, using the hash of the values of tags (`hub_account_id`, `hub_cluster_name`, and `managed_cluster_name`) on the role concatenated using `#` as the delimiter, in the alphabetical order of their keys:
+
+For example, given the following tags:
+- `hub_account_id: hubaccount1`
+- `hub_cluster_name: hubcluster1`
+- `managed_cluster_name: spokecluster1`
+
+The md5-hash-unique-identifier should be `hash(hubaccount1#hubcluster1#spokecluster1)` which is `1c502b1d59797fea758ae8177f246b2f`.
+
+#### Hub Cluster Prerequisites
+
+An IAM role named using following template:
+```
+arn:aws:iam::<hub-account-id>:role/<hub-cluster-name>_managed-cluster-identity-creator
+```
+
+Having the following access permissions policy:
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "iam:CreateRole",
+        "iam:DeleteRole",
+        "iam:UpdateAssumeRolePolicy",
+        "iam:PutRolePolicy",
+        "iam:DeleteRolePolicy",
+        "iam:AttachRolePolicy",
+        "iam:DetachRolePolicy",
+        "iam:CreatePolicy",
+        "iam:DeletePolicy",
+        "iam:CreatePolicyVersion",
+        "iam:DeletePolicyVersion",
+        "iam:ListAttachedRolePolicies",
+        "iam:ListRolePolicies",
+        "iam:ListPolicyVersions",
+        "iam:GetRole",
+        "iam:GetRolePolicy",
+        "iam:GetPolicy",
+        "iam:GetPolicyVersion",
+        "iam:PassRole"
+      ],
+      "Resource": [
+        "arn:aws:iam::<hub-account-id>:role/ocm-hub-*",
+        "arn:aws:iam::<hub-account-id>:policy/ocm-hub-*"
+      ]
+    }
+  ]
+}
+```
+
+And following trust relationship policy:
+```json
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::<hub-account-id>:oidc-provider/<oidc-provider>"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "<oidc-provider>:sub": "system:serviceaccount:open-cluster-management-hub:registration-controller-sa",
+          "<oidc-provider>:aud": "sts.amazonaws.com"
+        }
+      }
+    }
+  ]
+}
+```
+
+#### Cluster join initiated from managed cluster
+1. cluster-admin on managed cluster gets a bootstrap kubeconfig to connect to hub,
+   and deploy the agent on managed cluster.
+- it has the identity to create `ManagedCluster`.
+
+2. agent on managed cluster creates `ManagedCluster` on hub if it does not exist.
+- The name of `ManagedCluster` is passed using the `--cluster-name` option while running `clusteradm join` command.
+- The agent on the managed cluster will store it as `agent-name` in `Secret` `hub-kubeconfig-secret`, so restarting agent or redeploying agent will not lose the UID after the cluster is managed successfully.
```

Comment:
cluster name is stored as cluster-name, agent-name is an random uid which is unique for each agent.

Reply:
Thanks for spotting this, will update the proposal.